### PR TITLE
feat: Move `AnyReceipt` and `AnyHeader` to `alloy-consensus-any`

### PR DESCRIPTION
**Pull request by moricho**
## Motivation

ref https://github.com/alloy-rs/alloy/issues/1598

## PR Checklist

- [ ] Added Tests
- [ ] Added Documentation
- [ ] Breaking changes

### CODE REVIEW

File: crates/rpc-types-eth/src/transaction/receipt.rs [hunk 1]
```diff
@@ -138,7 +138,7 @@ impl<T> TransactionReceipt<T> {
 #[doc(alias = "AnyTxReceipt")]
 #[cfg(feature = "serde")]
 pub type AnyTransactionReceipt =
-    alloy_serde::WithOtherFields<TransactionReceipt<alloy_consensus::AnyReceiptEnvelope<Log>>>;
+    alloy_serde::WithOtherFields<TransactionReceipt<alloy_consensus_any::AnyReceiptEnvelope<Log>>>;
```
Comment (emhane) [inline crates/rpc-types-eth/src/transaction/receipt.rs:141]: make sense to move this receipt too right?

Comment (moricho) [reply crates/rpc-types-eth/src/transaction/receipt.rs]: ah you're right. moved it to `alloy-rpc-types-any` 🙏 
https://github.com/alloy-rs/alloy/pull/1609/commits/5b5d2ac2fd4044b1e63ba5fa52a17d5b46ab54d5

**Review by zerosnacks — APPROVED** [review]

File: crates/rpc-types-any/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+alloy-serde = { workspace = true, optional = true }
+
+[dev-dependencies]
+alloy-primitives.workspace = true
+
+serde.workspace = true
+serde_json.workspace = true
+
+[features]
+serde = ["dep:alloy-serde", "alloy-consensus-any/serde"]
```
Comment (klkvr) [inline crates/rpc-types-any/Cargo.toml:33]: pretty much all AnyNetwork types need WithOtherFields, so I think we don't need `serde` feature for this crate, let's just always include serde deps

File: crates/network/src/any/mod.rs [hunk 3]
```diff
@@ -6,14 +6,15 @@ pub use either::{AnyTxEnvelope, AnyTypedTransaction};
...
 
 use crate::Network;
-use alloy_rpc_types_eth::{AnyTransactionReceipt, Block, Transaction, TransactionRequest};
+use alloy_rpc_types_any::AnyTransactionReceipt;
+use alloy_rpc_types_eth::{Block, Transaction, TransactionRequest};
 use alloy_serde::WithOtherFields;
 
 /// A catch-all header type for handling headers on multiple networks.
-pub type AnyRpcHeader = alloy_rpc_types_eth::Header<alloy_consensus::AnyHeader>;
+pub type AnyRpcHeader = alloy_rpc_types_eth::Header<alloy_consensus_any::AnyHeader>;
```
Comment (klkvr) [inline crates/network/src/any/mod.rs:17]: let's move this alias to rpc-types-any

**Review by klkvr — CHANGES_REQUESTED** [review]

File: crates/rpc-types-any/Cargo.toml (hunk 2 above)
Comment (moricho) [reply crates/rpc-types-any/Cargo.toml]: Fixed at https://github.com/alloy-rs/alloy/pull/1609/commits/ea7e1f39801acf4c7ab7494922c3e75384827e08

File: crates/network/src/any/mod.rs (hunk 3 above)
Comment (moricho) [reply crates/network/src/any/mod.rs]: Fixed at https://github.com/alloy-rs/alloy/pull/1609/commits/da79415831a3af42cbaca63158aaa8aa484dbb07

File: crates/network/src/any/mod.rs [hunk 4]
```diff
@@ -6,15 +6,13 @@ pub use either::{AnyTxEnvelope, AnyTypedTransaction};
...
-use alloy_rpc_types_eth::{AnyTransactionReceipt, Block, Transaction, TransactionRequest};
+use alloy_rpc_types_any::{AnyRpcHeader, AnyTransactionReceipt};
+use alloy_rpc_types_eth::{Block, Transaction, TransactionRequest};
 use alloy_serde::WithOtherFields;
 
-/// A catch-all header type for handling headers on multiple networks.
-pub type AnyRpcHeader = alloy_rpc_types_eth::Header<alloy_consensus::AnyHeader>;
-
 /// A catch-all block type for handling blocks on multiple networks.
 pub type AnyRpcBlock =
```
Comment (klkvr) [inline crates/network/src/any/mod.rs:17]: `AnyRpcBlock` and `AnyRpcTransaction` aliases should also be in rpc-types-any

**Review by klkvr — APPROVED** [review]
lgtm, pending last nit on moving RPC aliases

**Pull request merged**